src/utils/components/CloudinitModal/utils/cloudinit-utils.ts

```diff
@@ -102,9 +102,20 @@ export const createDefaultCloudInitYAML = () => ({
   userData: '',
 });
 
+export const addCloudInitUpdateCMD = (userData: CloudInitUserData) => {
+  if (!userData.packages) {
+    userData.packages = ['dnf-automatic'];
```

Comment:
Maybe you could use simply `userData.packages ??= ['dnf-automatic'];` instead of this whole if block? See:
 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment
 
But be careful. If `userData.packages` can be an empty array here (and not null or undefined), I am not sure if we could use the nullish coalescing assignment here.

Same for `userData.runcmd` below.

Reply:
i will push those items if `runcmd` and `packages` are arrays

Reply:
Those check are not needed as you overriding any way and have a condition before calling this function.

BUT make sure you aren't ovverring when using dynamic ssh. If I remember correctly it's also adding value to runcmd

Reply:
I'll try it also with that